### ISSUE TITLE
segfault in user:portray on recursive terms

### ISSUE THREAD

**Issue by Gniourf** [SWI-Prolog/issues#32]
The following will segfault:

```
user:portray(t(X)) :- print(X).

tests(t(M),M,1).
tests(t(Rest),M,N) :- N>1, succ(N1,N), tests(Rest,M,N1).

main :-
   tests(A,'world',10000),
   format('Hello: ~p~n',[A]).
```

Bug occurs with SWI-Prolog 6.6.6 and 7.2.3 on Ubuntu 14.04 and Debian Jessie.

**Comment by JanWielemaker** [comment]
Is fixed with 9f1f5c9a09d3c25c46049a31c27c5bd729c84783.  Closing.

**Comment by logicmoo** [comment]
omg: This had been plaguing me for longer than I could remember!  I had
just assumed it was my fault :P
